### ISSUE TITLE
ANR com.facebook.jni.CppException 

### ISSUE THREAD

**Issue by devoren** [software-mansion/react-native-reanimated#4248]
### Description

We see several ANRs with resuscitation at the top of the stack trace. They also don't show up in bugsnag, only in the Play Store, which is now penalizing our app.
App crashes in production build: 

```
Exception com.facebook.jni.CppException: bad_weak_ptr
  at com.swmansion.reanimated.Scheduler.triggerUI
  at com.swmansion.reanimated.layoutReanimation.AnimationsManager.onViewCreate (AnimationsManager.java:135)
  at com.swmansion.reanimated.layoutReanimation.ReaLayoutAnimator.applyLayoutUpdate (ReanimatedNativeHierarchyManager.java:89)
  at com.facebook.react.uimanager.NativeViewHierarchyManager.updateLayout (NativeViewHierarchyManager.java:252)
  at com.facebook.react.uimanager.NativeViewHierarchyManager.updateLayout (NativeViewHierarchyManager.java:222)
  at com.facebook.react.uimanager.UIViewOperationQueue$UpdateLayoutOperation.execute (UIViewOperationQueue.java:169)
  at com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:915)
  at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:1026)
  at com.facebook.react.uimanager.UIViewOperationQueue.access$2600 (UIViewOperationQueue.java:47)
  at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:1086)
  at com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:29)
  at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:175)
  at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:85)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:1090)
  at android.view.Choreographer.doCallbacks (Choreographer.java:888)
  at android.view.Choreographer.doFrame (Choreographer.java:816)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:1078)
  at android.os.Handler.handleCallback (Handler.java:873)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:201)
  at android.app.ActivityThread.main (ActivityThread.java:6810)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:873)
```

### Steps to reproduce

After using the library in production

### Snack or a link to a repository

NA

### Reanimated version

^2.14.4

### React Native version

0.71.4

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Release mode

### Device

Real device

### Device model

HUAWEI HWANE, xiaomi lavender ANDROID 9 SDK 28

### Acknowledgements

Yes

**Comment by devoren** [comment]
Also another crash on 
POCO veux (POCO X4 Pro 5G) - Android 11 (SDK 30): 
```
Exception com.facebook.jni.CppException: bad_weak_ptr
  at com.swmansion.reanimated.Scheduler.triggerUI
  at com.swmansion.reanimated.Scheduler$1.run (Scheduler.java:24)
  at com.swmansion.reanimated.Scheduler$2.runGuarded (Scheduler.java:43)
  at com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:30)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8163)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:656)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:972)
```

**Comment by devoren** [comment]
Last crash: 
```
java.lang.RuntimeException
Exception java.lang.RuntimeException: Cannot read property 'current' of undefined

TypeError: Cannot read property 'current' of undefined
    at workletValueSetter (/Users/.../app/node_modules/react-native-reanimated/src/reanimated2/core.ts (143:0):1:384)
  at com.swmansion.reanimated.Scheduler.triggerUI
  at com.swmansion.reanimated.Scheduler$1.run (Scheduler.java:24)
  at com.swmansion.reanimated.Scheduler$2.runGuarded (Scheduler.java:43)
  at com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:30)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8669)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:571)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1135)
```

**Comment by devoren** [comment]
@maulik54e These problems appear in the play console (Release stability)

**Comment by robwalkerco** [comment]
I'm also seeing these same crashes.
I haven't tested yet, but there is a fix for a ANR in https://github.com/software-mansion/react-native-reanimated/releases/tag/2.15.0 so I'm wondering if that is fixing this same issue?

**Comment by robinvk6** [comment]
We are on 2.15.0 and we continue to see the issue.

Affected devices -
Samsung SM-S908U
Samsung SM-S918U
Samsung SM-F711U
Samsung SM-S901U
Samsung SM-S916U
Coolpad CP3706AS
Samsung SM-F926U
Samsung SM-G781V
Samsung SM-G970U

**Comment by markrickert** [comment]
I'm seeing this same issue on 2.17.0 using react-native 0.71.12. The crash report was from a Samsung Galaxy A8(2018) running Android 9, so i can't even try and reproduce it, unfortunately. I don't have any android devices that old anymore.

```
Fatal Exception: java.lang.RuntimeException
Cannot read property 'current' of undefined TypeError: Cannot read property 'current' of undefined at workletValueSetter (/Users/appname/actions-runner/_work/app/app/node_modules/react-native-reanimated/src/reanimated2/core.ts (143:0):1:384)
com.swmansion.reanimated.Scheduler.triggerUI (Scheduler.java)
com.swmansion.reanimated.Scheduler$1.run (Scheduler.java:24)
com.swmansion.reanimated.Scheduler$2.runGuarded (Scheduler.java:43)
com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:30)
android.os.Handler.handleCallback (Handler.java:873)
android.os.Handler.dispatchMessage (Handler.java:99)
android.os.Looper.loop (Looper.java:214)
android.app.ActivityThread.main (ActivityThread.java:7078)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:494)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:964)
```

**Comment by Latropos** [comment]
Closing since fix #4239 is merged
